contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,305 @@
+                "_amount above available sOHM"
+            );
+
+            borrower[msg.sender].collateralInSOHM -= _amount;
+            borrower[msg.sender].collateralInGOHM -= IgOHM(gOHM).balanceTo(_amount);
+
+            IERC20(sOHM).safeTransfer(_to, _amount);
+        }
+    }
+
+    /**
+     * @notice repay debt with collateral
+     * - msg.sender must be a borrower
+     * - msg.sender's OHM allowance must be >= _ohmAmount
+     */
+    function repayDebtWithCollateral() external {
+        uint256 depositedCollateralAfterRepay = _repay(msg.sender);
+        totalOutstandingDebt -= borrower[msg.sender].debt;
+
+        borrower[msg.sender].collateralInSOHM = depositedCollateralAfterRepay;
+        borrower[msg.sender].collateralInGOHM = IgOHM(gOHM).balanceTo(depositedCollateralAfterRepay);
+
+        borrower[msg.sender].debt = 0;
+    }
+
+    /**
+     * @notice repay debt with collateral and withdraw the leftover
+     * - msg.sender must be a borrower
+     * - msg.sender's OHM allowance must be >= _ohmAmount
+     */
+    function repayDebtWithCollateralAndWithdrawTheRest(address _tokenToReceiveExcess) external {
+        uint256 depositedCollateralAfterRepay = _repay(msg.sender);
+
+        if (depositedCollateralAfterRepay > 0) {
+            if (_tokenToReceiveExcess == sOHM) {
+                IERC20(sOHM).safeTransfer(msg.sender, depositedCollateralAfterRepay);
+            } else if (_tokenToReceiveExcess == gOHM) {
+                IStaking(staking).wrap(msg.sender, depositedCollateralAfterRepay);
+            }
+        }
+
+        totalOutstandingDebt -= borrower[msg.sender].debt;
+        updateBorrowerStats(msg.sender);
+    }
+
+    /**
+     * @notice deposits OHM to pay debt
+     * - msg.sender must be a borrower
+     * - msg.sender's OHM allowance must be >= _ohmAmount
+     * @param _ohmAmount amount of OHM to borrow
+     */
+    function repayDebtWithOHM(uint256 _ohmAmount) external isBorrower(msg.sender) {
+        require(borrower[msg.sender].debt > 0, "does not have outstanding debt");
+
+        totalOutstandingDebt -= _ohmAmount;
+        borrower[msg.sender].debt = borrower[msg.sender].debt - _ohmAmount;
+
+        IERC20(OHM).safeTransferFrom(msg.sender, address(this), _ohmAmount);
+        ITreasury(treasury).repayDebtWithOHM(_ohmAmount);
+    }
+
+    /**
+     * @notice repays debt using collateral and returns remaining tokens to owner
+     * - onlyOwner (or governance)
+     * - sends remaining tokens to owner in sOHM
+     * @param _account the address that will interact with contract
+     * @param _to where to send remaining gOHM
+     */
+    function forceRepay(address _account, address _to) external onlyGovernor {
+        uint256 collateralAfterDebtPayment = _repay(_account);
+        totalOutstandingDebt -= borrower[_account].debt;
+
+        updateBorrowerStats(_account);
+        if (collateralAfterDebtPayment > 0) IERC20(sOHM).safeTransfer(_to, collateralAfterDebtPayment);
```

Comment:
_repay() unstakes sOHM into OHM. Following lines transfer sOHM or wrap into gOHM, but you have OHM now?